### PR TITLE
EXP-2339 Generate Swift code to override a component feature config

### PR DESCRIPTION
**Pull request by jhugman**
Fixes [EXP-2339](https://mozilla-hub.atlassian.net/browse/EXP-2339).

This is the Swift equivalent of [EXP-2338](https://mozilla-hub.atlassian.net/browse/EXP-2338).

### Pull Request checklist ###

- [x] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes a changelog entry in [CHANGES_UNRELEASED.md](../CHANGES_UNRELEASED.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[ac: android-components-branch-name]` and/or `[fenix: fenix-branch-name]` to the PR title.

### CODE REVIEW

File: components/support/nimbus-fml/src/parser.rs [hunk 1]
```diff
@@ -112,7 +114,11 @@ pub(crate) struct ImportBlock {
 #[serde(deny_unknown_fields)]
 pub(crate) struct FeatureBody {
     description: String,
-    variables: HashMap<String, FieldBody>,
+    // We need these in a deterministic order, so they are stable across multiple
+    // runs of the same manifests:
+    // 1. Swift insists on args in the same order they were declared.
+    // 2. imported features are declared and constructed in different runs of the tool.
+    variables: BTreeMap<String, FieldBody>,
```
Comment (travis79) [inline components/support/nimbus-fml/src/parser.rs:121]: Does this struct map through UniFFI to Swift? If so, what type do you get out of the BTreeMap on the other side? I'm recalling problems with dictionaries in Swift being non-deterministically ordered, but that may not be a problem here.

**Review by travis79 — APPROVED** [review]
Nothing serious to note, just a question around the `FeatureBody`

File: components/support/nimbus-fml/src/parser.rs (hunk 1 above)
Comment (jhugman) [reply components/support/nimbus-fml/src/parser.rs:121]: No, none of this code runs on device, and the code it generates is all consuming JSON objects coming through the FFI— i.e. that code doesn't correspond to anything uniffi.

Comment (travis79) [reply components/support/nimbus-fml/src/parser.rs:121]: Right, that's what I expected. I was just recalling some interesting bugs around Swift dictionaries.  Carry on :)

**Pull request merged**